### PR TITLE
feat: update to  ticks

### PR DESCRIPTION
**Pull request by peterMangrove**
This PR tries to update the SDK to use the new Mangrove version with ticks

Tasks:

- [x] Get code to compile
- [x] Get tests to compile
- [ ] Unit tests green
- - [ ] Kandel unit tests
- - [x] Other unit tests
- [ ] Integration tests green
- - [ ] Kandel tests
- - [x] Other tests 
- [ ] Go through FIXMEs
- [ ] Review code
- [ ] Add documenation

### CODE REVIEW

File: src/semibook.ts [hunk 1]
```diff
@@ -511,14 +502,14 @@ class Semibook
...
-      local: { offer_gasbase },
+      _local: { kilo_offer_gasbase },
     } = await this.getRawConfig();
 
     // Assume up to offer_gasbase is used also for the bad price call, and
     // the last offer (which could be first, if taking little) needs up to gasreq*64/63 for makerPosthook
     const gas = res.totalGasreq
       .add(BigNumber.from(res.lastGasreq).div(63))
-      .add(offer_gasbase.mul(Math.max(res.offersConsidered, 1)));
+      .add(kilo_offer_gasbase.mul(Math.max(res.offersConsidered, 1)));
```
Comment (peterMangrove) [inline src/semibook.ts:512]: Should multiply by 1k

File: src/semibook.ts [hunk 2]
```diff
@@ -899,7 +896,7 @@ class Semibook
       }
 
       case "SetGasbase":
-        this.#offer_gasbase = event.args.offer_gasbase.toNumber();
+        this.#kilo_offer_gasbase = event.args.offer_gasbase.toNumber();
```
Comment (peterMangrove) [inline src/semibook.ts:899]: event.args.offer_gasbase returns full gabase, not kilo gasbase

File: src/util/forgeScript.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ export async function runScript(params: {
         --private-key ${params.mnemonic.key(0)} `
         : ""
     } \
-    --broadcast -vvv \
```
Comment (peterMangrove) [inline src/util/forgeScript.ts:81]: dont remove

File: src/util/test/mgvIntegrationTestUtil.ts [hunk 4]
```diff
@@ -367,9 +368,9 @@ export const postNewOffer = async ({
   market,
   ba,
   maker,
-  wants = 1,
+  tick = 1,
   gives = "90000000000000000",
-  gasreq = 5e4,
+  gasreq = 10e4, // FIXME: what is the correct gasreq?
```
Comment (peterMangrove) [inline src/util/test/mgvIntegrationTestUtil.ts:373]:
```suggestion
  gasreq = 1e5,
```

File: src/util/trade.ts [hunk 5]
```diff
@@ -30,31 +34,51 @@ class Trade {
...
+      );
+      const priceWithSlippage = priceWithCorrectDecimals
+        .mul(100 + slippage)
+        .div(100);
       if ("volume" in params) {
-        wants = Big(params.volume);
-        gives = wants.mul(params.price);
+        fillVolume = Big(params.volume);
+        if (params.price == 0) {
+          tick = BigNumber.from(MIN_TICK);
```
Comment (peterMangrove) [inline src/util/trade.ts:49]: Should be MAX?

File: src/util/trade.ts [hunk 6]
```diff
@@ -64,32 +88,50 @@ class Trade {
...
+        } else {
+          tick = TickLib.getTickFromPrice(priceWithSlippage);
+        }
         fillWants = false;
       } else {
-        wants = Big(params.total);
-        gives = wants.div(params.price);
+        fillVolume = Big(params.total);
+        if (params.price == 0) {
+          tick = BigNumber.from(MIN_TICK);
```
Comment (peterMangrove) [inline src/util/trade.ts:111]: Should be MAX?

File: src/util/tradeEventManagement.ts [hunk 7]
```diff
@@ -44,23 +46,23 @@ class TradeEventManagement {
...
-      prev: this.#rawIdToId(raw.prev),
       gasprice: raw.gasprice.toNumber(),
       maker: raw.maker,
       gasreq: raw.gasreq.toNumber(),
+      tick: raw.tick,
       gives: gives,
-      wants: wants,
-      volume: baseVolume,
-      price: price,
+      price: TickLib.priceFromTick(raw.tick)[divOrMul](
```
Comment (peterMangrove) [inline src/util/tradeEventManagement.ts:63]: Consider creating helper functions for getting readable price from tick

**Review by dontrolle — APPROVED** [review]
✅

**Pull request merged**